extensions/blocks/social-previews/social-previews.php

```diff
@@ -12,17 +12,17 @@
 use Jetpack_Gutenberg;
 
 const FEATURE_NAME = 'social-previews';
-const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+const EXTENSION_NAME   = 'jetpack/' . FEATURE_NAME;
 
 /**
  * Registers the Social Previews feature with the block editor.
  */
-function register_block() {
-	jetpack_register_block(
-		BLOCK_NAME,
+function register_extension() {
+	jetpack_register_extension(
+		EXTENSION_NAME,
 		array(
 			'plan_check' => true,
```

Comment:
I realise it's not for this change, but the idea with `plan_check` was that it was a way of opting into the plan check until we ensured that the features for each plan were set up correctly for all the blocks/extensions.

As this file only exists to register the extension it would be good to take this a step further, and blocks and extensions auto-registered, maybe using something similar to block.json?